docs/tasks/administer-cluster/kubeadm-upgrade-1-7.md

```diff
@@ -0,0 +1,59 @@
+---
+assignees:
+- pipejakob
+title: Upgrading kubeadm clusters from 1.6 to 1.7
+redirect_from:
+- "/docs/admin/kubeadm-upgrade-1-7/"
+- "/docs/admin/kubeadm-upgrade-1-7.html"
+---
+
+* TOC
+{:toc}
+
+This guide is for upgrading kubeadm clusters from version 1.6.x to 1.7.x.
+Upgrades are not supported for clusters lower than 1.6, which is when kubeadm
+became Beta.
+
+**WARNING**: These instructions will **overwrite** all of the resources managed
+by kubeadm (static pod manifest files, service accounts and RBAC rules in the
+`kube-system` namespace, etc.), so any customizations you may have made to these
+resources after cluster setup will need to be reapplied after the upgrade. The
+upgrade will not disturb other static pod manifest files or objects outside the
+`kube-system` namespace.
+
+1. Back up `/etc/kubernetes`.
+
+       sudo cp -r /etc/kubernetes /root/kubernetes-backup
+
+2. Upgrade system packages.
+
+   Upgrade your OS packages for kubectl, kubeadm, kubelet, and kubernetes-cni.
+
+   a. On Debian, this can be accomplished with:
+
+       sudo apt-get update
+       sudo apt-get upgrade
+
+   b. On CentOS/Fedora, you would instead run:
+
+       sudo yum update
+       sudo yum upgrade
+
+3. Restart kubelet.
+
+       sudo systemctl restart kubelet
+
+4. Perform kubeadm upgrade.
+
+       sudo kubeadm init --skip-preflight-checks --kubernetes-version <DESIRED_VERSION>
+
+   For instance, if you want to upgrade to `1.7.0`, you would run:
+
+       sudo kubeadm init --skip-preflight-checks --kubernetes-version v1.7.0
+
+5. Upgrade CNI provider.
```

Comment:
Done. However, the 1.7.0-beta.2 version doesn't seem to have the `kube-proxy` regression fix, so the pod ends up crash looping. I tested with kubeadm built from the HEAD of `release-1.7` and the steps worked fine.

The next 1.7 beta release is due out today, so I'll be able to build new debs/rpms for it and post them for testing. I'll also start building them for the HEAD of `release-1.7` so people can still test these instructions ASAP.